Autopilot/PathManager/Inc/pathDatatypes.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef PATH_DATATYPES_HPP
+#define PATH_DATATYPES_HPP
+
+/********************************************************************
+* Enums
+********************************************************************/
+
+enum _ModifyFlightPathCommand { NO_FLIGHT_PATH_EDIT = 0, INITIALIZE_FLIGHT_PATH, APPEND, INSERT, UPDATE, DELETE, NUKE }; // Used by cruisingState
+enum _GetNextDirectionsCommand { REGULAR_PATH_FOLLOWING = 0, TOGGLE_HOLDING, TOGGLE_HEAD_HOME }; // Used by cruisingState
+
+/********************************************************************
+* Imporatant Structs
+********************************************************************/
+
+struct Telemetry_Waypoint_Data_t {
+    long double latitude;
+    long double longitude;
+    int altitude;
+    float turnRadius;
+    int waypointType; // 0 = Path follow, 1 = Orbit, 2 = Hold
+};
+
+struct Telemetry_PIGO_t {
+    /* Parameters for the waypoint manager (crusingState) */
+    int numWaypoints;
+    
+    _ModifyFlightPathCommand waypointModifyFlightPathCommand; 
+    bool initializingHomeBase; // 0 = no, 1 = yes
+    _GetNextDirectionsCommand waypointNextDirectionsCommand; 
+    int holdingAltitude;
+    int holdingTurnRadius;
+    char holdingTurnDirection; // 0 = CW, 1 = CCW
+
+    // When modifying the flight path.
+    int nextId;
+    int prevId;
+    int modifyId;
+
+    Telemetry_Waypoint_Data_t * waypoints[100]; // Somehow need to get PATH_BUFFER_SIZE here...
```

Comment:
THis is an array of pointers atm. Doesn't make sense to me. When the waypoints come in from the ground, they should just be packed into this struct, no ? Or do they need to be allocated on the heap for some reason ?

Reply:
Oh my bad. I forgot that statically allocated arrays don't have the pointer.